starlette/requests.py

```diff
@@ -125,6 +126,12 @@ def url_for(self, name: str, **path_params: typing.Any) -> str:
     async def json(self) -> typing.Any:
         if not hasattr(self, "_json"):
             body = await self.body()
+            if body is b"":
+                if "app" in self._scope:
```

Comment:
I’m planning to get rid of this approach and just raise HTTPExcpetion in the router/endpoint implementations. The low level interface will document how to catch the exception or override the behaviour.